File: Password Factory/Controls/StrengthMeter.m
//
//  StrengthMeter.m
//  Password Factory
//
//  Created by Cristiana Yambo on 8/13/15.
//  Copyright (c) 2015 c13. All rights reserved.
//

#import "StrengthMeter.h"
#import "StyleKit.h"
#import "Utilities.h"
@implementation StrengthMeter

- (void)drawRect:(NSRect)dirtyRect {
    [super drawRect:dirtyRect];
    float strength = self.strength + 0.01; // give a little size to zero strength
    [StyleKit drawStrengthMeterWithStrengthColor:[self getStrengthColor] strength:strength size:dirtyRect.size];
}
-(void)viewWillDraw {
    if ([Utilities isDarkMode]) {
        [self setAlphaValue:0.25];
    } else {
        [self setAlphaValue:1];
    }
}
@end

File: Password Factory iOS/Controls/StrengthMeter.swift
//
//  StrengthMeter.swift
//  Password Factory iOS
//
//  Created by Cristiana Yambo on 12/2/17.
//  Copyright © 2017 Cristiana Yambo. All rights reserved.
//

import UIKit
@IBDesignable

/// Strength Meter View
class StrengthMeter: UIView {
    var strength = 1.0
    override func draw(_ rect: CGRect) {
        super.draw(rect)
        StyleKit.drawStrengthMeter(frame: rect, resizing: .stretch, strengthColor: ColorUtilities.getStrengthColor(Float(strength)), strength: CGFloat(strength))
    }
    override func prepareForInterfaceBuilder() {
        super.prepareForInterfaceBuilder()
        updateStrength(s: 0.5)
    }
    override func layoutSubviews() {
        super.layoutSubviews()
        setDisplay()
    }
    
    /// Update the strength, and will redraw meter
    ///
    /// - Parameter s: strength to set
    func updateStrength(s: Double) {
        
        strength = s
        if (strength < 0.0) { strength = 0.0 }
        if (strength > 1.0) { strength = 1.0 }
        setNeedsDisplay()
    }
    func setDisplay() {
        addBorder([.top,.bottom], color: PFConstants.containerBorderColor)

    }
}

